src/main/java/Parser.java

```diff
@@ -0,0 +1,170 @@
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+
+public class Parser {
+
+    /**
+     * Returns the task number (as indexed in the task list)
+     * of the task to be deleted or marked as done.
+     *
+     * @param keyword indicates if the task is to be deleted or marked as done.
+     * @param task task to be deleted or marked as done (required).
+     * @return index of the required task in the task list.
+     */
+    private static int getTaskNumber(String keyword, String task) {
+        int idx = 0;
+        for (int i = task.length() - 1; i > keyword.length(); i--) {
+            idx+= (task.charAt(i) - 48) * Math.pow(10, task.length() - 1 - i);
+        }
+        return idx - 1;
+    }
+
+    /**
+     * Returns the index of the task to be deleted or marked as done
+     * in case it is valid, else throws a BlitzException.
+     *
+     * @param command command entered by the user.
+     * @param listSize current size of the task list.
+     * @return index of the required task in the task list if valid.
+     * @throws BlitzException if the computed index is invalid.
+     */
+    public static int getIndex(String command, int listSize) throws BlitzException {
+        String keyword = command.substring(0, command.indexOf(' '));
+        int index = getTaskNumber(keyword,command);
+        if (index < 0 || index >= listSize) {
+            throw new BlitzException("You are attempting to "
+                + (keyword.equals("done") ? "mark" : "delete")
+                    + " an invalid task number!");
+        }
+        return index;
+    }
+
+    /**
+     * Makes sense of the user command and performs the
+     * required action.
+     *
+     * @param command command entered by the user.
+     * @param tasks current list of tasks
+     * @param ui current user interface instance
+     * @throws BlitzException if the command description is missing, or
+     *     if a request to print an empty list is made or if the user enters
+     *     an invalid command.
+     */
+    public static void parse(String command, TaskList tasks, Ui ui) throws BlitzException{
```

Comment:
Your parse method is really long. Maybe you could try to abstract some of the code out into helper methods?